include/nbl/asset/utils/CGLSLCompiler.h

```diff
@@ -0,0 +1,174 @@
+// Copyright (C) 2018-2022 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_ASSET_C_GLSL_COMPILER_H_INCLUDED_
+#define _NBL_ASSET_C_GLSL_COMPILER_H_INCLUDED_
+
+#include "nbl/asset/utils/ISPIRVOptimizer.h"
+#include "nbl/asset/utils/IShaderCompiler.h"
+
+namespace nbl::asset
+{
+
+class NBL_API2 CGLSLCompiler final : public IShaderCompiler
+{
+	public:
+		IShader::E_CONTENT_TYPE getCodeContentType() const override { return IShader::E_CONTENT_TYPE::ECT_GLSL; };
+
+		CGLSLCompiler(core::smart_refctd_ptr<system::ISystem>&& system);
+
+		struct SOptions : IShaderCompiler::SOptions
+		{
+			void setCommonData(const IShaderCompiler::SOptions& opt)
+			{
+				static_cast<IShaderCompiler::SOptions&>(*this) = opt;
+			}
+
+			virtual IShader::E_CONTENT_TYPE getCodeContentType() const override { return IShader::E_CONTENT_TYPE::ECT_GLSL; };
+		};
+
+		/**
+		If options.stage is ESS_UNKNOWN, then compiler will try to deduce shader stage from #pragma annotation, i.e.:
+		#pragma shader_stage(vertex),       or
+		#pragma shader_stage(tesscontrol),  or
+		#pragma shader_stage(tesseval),     or
+		#pragma shader_stage(geometry),     or
+		#pragma shader_stage(fragment),     or
+		#pragma shader_stage(compute)
+
+		Such annotation should be placed right after #version directive.
+
+		This function does NOT process #include directives! Use resolveIncludeDirectives() first.
+
+		@params code high level code
+		@params options see IShaderCompiler::SOptions
+			- entryPoint Must be "main" since shaderc does not allow other entry points for GLSL. Kept with hope that shaderc will drop that requirement.
+
+		@returns Shader containing SPIR-V bytecode.
+		*/
+		core::smart_refctd_ptr<ICPUBuffer> compileToSPIRV(const char* code, const CGLSLCompiler::SOptions& options) const;
+
+		core::smart_refctd_ptr<ICPUShader> createSPIRVShader(const char* code, const CGLSLCompiler::SOptions& options) const;
```

Comment:
why do we have one for the buffer and one for the shader output?

Reply:
these functions map to previous `compileSPIRVFromGLSL` and `createSPIRVFromGLSL`
Sometimes I think sometimes we use ICPUShader and ICPUBuffer interchangabily. I had plans in mind to go over all these and see if I can clean things up.
(but sometimes ICPUShader is returned because stage will be inferred from the code using pragmas, but not in this case)

Reply:
Anyways, 99% of the codebase uses the Buffer one so If I merge them into one function that returns ICPUShader I'll have to touch up many places

Reply:
but now I investigated a bit and It seems I can merge them into one to return ICPUShader no problem

Reply:
then do that